vllm/v1/core/sched/scheduler.py

```diff
@@ -1154,9 +1155,22 @@ def update_draft_token_ids(
             else:
                 request.spec_token_ids = spec_token_ids
 
-    def get_request_counts(self) -> tuple[int, int]:
-        """Returns (num_running_reqs, num_waiting_reqs)."""
-        return len(self.running), len(self.waiting)
+    def _get_corrupted_requests_count(self, include_corrupted: bool) -> int:
+        """Get the count of corrupted requests if enabled, otherwise return 0.
+
+        This method centralizes the corrupted requests counting logic to avoid
+        code duplication and improve performance.
+        """
+        if not include_corrupted:
+            return 0
+        return sum(req.is_output_corrupted for req in self.running)
```

Comment:
I really don't like that we're doing this twice per step (AFAICT), but that's the fault of how `_maybe_publish_request_counts()` was implemented. IMHO, it should use the `SchedulerStats` that would have been prepared already